pkg/controller/crdmirroring/crdmirroring_controller_test.go

```diff
@@ -0,0 +1,1606 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crdmirroring
+
+import (
+	"context"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+
+	crdv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/crd/v1alpha1"
+	crdv1alpha2 "github.com/vmware-tanzu/antrea/pkg/apis/crd/v1alpha2"
+	crdclientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	fakeversioned "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned/fake"
+	crdinformers "github.com/vmware-tanzu/antrea/pkg/client/informers/externalversions"
+	crdv1a1lister "github.com/vmware-tanzu/antrea/pkg/client/listers/crd/v1alpha1"
+	crdv1a2lister "github.com/vmware-tanzu/antrea/pkg/client/listers/crd/v1alpha2"
+	"github.com/vmware-tanzu/antrea/pkg/controller/crdmirroring/crdhandler"
+	"github.com/vmware-tanzu/antrea/pkg/controller/crdmirroring/types"
+	legacycore "github.com/vmware-tanzu/antrea/pkg/legacyapis/core/v1alpha2"
+	legacyops "github.com/vmware-tanzu/antrea/pkg/legacyapis/ops/v1alpha1"
+	legacysecurity "github.com/vmware-tanzu/antrea/pkg/legacyapis/security/v1alpha1"
+	legacycrdclientset "github.com/vmware-tanzu/antrea/pkg/legacyclient/clientset/versioned"
+	legacyfakeversioned "github.com/vmware-tanzu/antrea/pkg/legacyclient/clientset/versioned/fake"
+	legacycrdinformers "github.com/vmware-tanzu/antrea/pkg/legacyclient/informers/externalversions"
+	legacycorelister "github.com/vmware-tanzu/antrea/pkg/legacyclient/listers/core/v1alpha2"
+	legacyopslister "github.com/vmware-tanzu/antrea/pkg/legacyclient/listers/ops/v1alpha1"
+	legacysecuritylister "github.com/vmware-tanzu/antrea/pkg/legacyclient/listers/security/v1alpha1"
+)
+
+const (
+	informerDefaultResync = 30 * time.Second
+	timeout               = 2 * time.Minute
```

Comment:
2 minute can make the test execute very long if the test is broken, it typically takes less than 1 second, right? Change it to 2 seconds at most?

Reply:
2 seconds is ok.